src/ORM/Table.php

```diff
@@ -1175,6 +1175,32 @@ public function belongsToMany($associated, array $options = [])
      */
     public function find($type = 'all', $options = [])
     {
+        $deprecatedKeys = [
+            // 'fields', // 24 error warnings
+            // 'conditions', // 77 errors warnings, core seems to rely on thi...
+            'join', // 0 error warnings
+            // 'order', // 19 error warnings
+            // 'limit', // 21 error warnings
+            'offset', // 0 error warnings
+            // 'group', // 1 error warning in cakephp/src/Controller/Component/PaginatorComponent.php, line: 201
+            'having', // 0 error warnings
+            // 'contain', // 83 error warnings, core seems to rely on this...
+            // 'page', // 17 error warnings
```

Comment:
If we want to remove the 2.x style find() we should go through the code and replace this usage with query builder calls. Partially deprecating the 2.x style find() doesn't make sense to me.

Reply:
Can we convert array to method ?or  convert 2x style to 3x internally?

Reply:
@markstory 
Cc @lorenzo

Reply:
@markstory while I agree that seems to be a major undertaking. I can go one by one key and see how far I can push it starting with the keys that only throw some error warnings if the consensus/decision is to deprecate the whole 2.x style finders including the way `Table::get()` works. Ref: https://book.cakephp.org/3.0/en/orm/retrieving-data-and-resultsets.html#getting-a-single-entity-by-primary-key

Reply:
What about making `get()` chain to the query object similar to `firstOrFail()`?

Current 3.x Style `get()`:
```php
$article = $this->Articles->get($id, [
    'finder' => 'translations',
    'contain' => ['Comments' => 'Users'],
    'cache' => 'custom','key' => 'mykey']);
```

Proposed 4.x style `get()`:
```php
$article = $this->Articles
    ->find('translations')
    ->contain('Comments.Users')
    ->get([$id], ['cache' => 'custom', 'key' => 'mykey']);
```

Reply:
That could work. I think get() as a finder method could have strange interactions with finders that set conditions. The `get()` method could convert those specific options into query builder method calls too.

Reply:
`get()` could just accept a query builder as an option or a second argument.

```php
$this->Articles->get($id, [
    'foo' => 'bar'
    'query' => function($q) {
        return $q->contain('Comments');
    }
]);
```
or:
```php
$this->Articles->get($id,  function($q) {
    return $q->applyOptions([
        'foo' => 'bar'
    ])->contain('Comments');
});
```

Reply:
I like the 
```
->get($id,function($q){
return $q->order(...)
}
```

Reply:
>If we want to remove the 2.x style find() we should go through the code and replace this usage with query builder calls

I see the core ,internally all of these key will convert to method 
https://github.com/cakephp/cakephp/blob/0e493afa03d656ca7dcbfe63056ce167a5d5fb3d/src/ORM/Query.php#L756

so 
```

Reply:
if you say replace all usage of array style(within core) to method that is another thing